src/cairoUtilFuncGen/storage/storageToMemory.ts

```diff
@@ -96,8 +102,15 @@ export class StorageToMemoryGen extends StringIndexedFuncGen {
               ];
             } else {
               const funcName = this.getOrCreate(copyType);
+              const copy =
+                copyType instanceof ArrayType && copyType.size === undefined
+                  ? [
+                      `let (dyn_loc) = ${this.storageReadGen.genFuncName(copyType)}(loc)`,
```

Comment:
Is there a reason for doing this with a readGen rather than with a WARP_STORAGE.read?